compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -76,6 +78,19 @@ pub fn write_output_file(
                 file_type,
             )
         };
+
+        // Record artifact sizes for self-profiling
+        if result == llvm::LLVMRustResult::Success {
+            let artifact_kind = match file_type {
+                llvm::FileType::ObjectFile => "object_file",
+                llvm::FileType::AssemblyFile => "assembly_file",
+            };
+            record_artifact_size(self_profiler_ref, artifact_kind, output);
+            if let Some(dwo_file) = dwo_output {
+                record_artifact_size(self_profiler_ref, "dwo_file", dwo_file);
+            }
+        }
```

Comment:
Can you update https://github.com/rust-lang/rust/blob/60952bc3da442cf2ea11dcc9c80a461a659bbcd7/compiler/rustc_codegen_cranelift/src/driver/aot.rs#L39 too please?

Reply:
Done in bjorn3/rustc_codegen_cranelift@f3b2d37e7754458a14ac99cee970ede383eddc2f.